lib/isorender/dom-shims.js

```diff
@@ -34,25 +34,52 @@ global.requestAnimationFrame = global.requestAnimationFrame || requestAnimationF
     const child = origFunc.apply(this, arguments);
     requestAnimationFrame(() => {
       if (!child.initialized && child.connectedCallback) {
+        child.isConnected = true;
         child.connectedCallback();
       }
     });
   };
 });
 
+// patch removeChild to call disconnectedCallback
+const origRemoveChild = Element.prototype.removeChild;
+Element.prototype.removeChild = function(child) {
+  origRemoveChild.call(this, child);
+  const childNode = child;
+  if (childNode && childNode.disconnectedCallback && childNode.initialized) {
+    childNode.isConnected = false;
+    childNode.disconnectedCallback();
+  }
+  return child;
+};
+
+Node.DOCUMENT_FRAGMENT_NODE = 11
+
+// html-element does not provide hasAttribute
+Element.prototype.hasAttribute = function (name) {
```

Comment:
just be consistent with the rest of the file
```suggestion
Element.prototype.hasAttribute = function(name) {
```

Reply:
(PS how is prettier not taking care of this for us?)

Reply:
sorry, I misunderstood the previous suggestion. happy to fix the eslint/prettier autofix setup here or in a later PR

Reply:
yeah no need to block this one on prettier shenanigans